napi_js_props.md

```diff
@@ -0,0 +1,225 @@
+# N-API - JavaScript Data Types and Values 
+
+N-API exposes a set of APIs to interrogate and set properties on JavaScript objects.
+Some of these types are documented under [Section 7](https://tc39.github.io/ecma262/#sec-operations-on-objects) 
+of the [ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+
+Fundamentally, all properties in N-API can be represented in one of the following forms:
+- Named: a simple utf8 encoded string
+- Integer-Indexed: an index value represented by `uint32_t`
+- JavaScript Object: a complex JavaScript Object represented by `napi_value`
```

Comment:
Not sure if you're referring to the JavaScript Object case, but it's for when a property key is not a string or a number- added a sentence to clarify